r2/r2/public/static/js/reddit.js

```diff
@@ -1229,6 +1229,17 @@ $(function() {
             $("#moresearchinfo").slideUp();
             event.preventDefault();
         });
+
+        /* Add a scheme to URL input fields if not already present */
+        $("input[type=url]").blur(function() {
```

Comment:
Hmm, I don't know if we want to do this everywhere - it seems like a reasonable thing to do, but that sort of wide-reaching change also has a tendency to unexpectedly break things.  It might be best to start by targeting the submission page specifically, then widening it as we find more areas it'd be useful.